### DISCUSSION TITLE
List of items in 2 columns. Split elements.

### DISCUSSION THREAD

**Discussion by greenpau** [tailwindlabs/tailwindcss#13044] [Help] [answered]
I am creating 2 column layout for large displays and 1 column layout for anything smaller.

```
export const ServiceSelector: React.FC<TrainingContentProps> = (props) => {
  return (
    <div id="getting-started" className="relative isolate">
      <h2 className="sr-only">Getting Started</h2>
      <div className="lg:columns-2 lg:gap-16 mx-auto pt-12 mt-16 pb-20 sm:mt-20 lg:px-32">
        <ul role="list" className="-mt-6 divide-y divide-gray-900/5 border-b border-gray-900/5">
          {props.sections &&
            props.sections.map((item, itemIdx) => (
              <li key={`section-even-${itemIdx}`} className="relative flex gap-x-6 py-6 px-12 hover:bg-gray-50">
                <div className="flex h-11 w-11 flex-none items-center justify-center rounded-lg bg-gray-50">
                  <item.icon className="h-6 w-6 text-gray-600" aria-hidden="true" />
                </div>
                <div className="flex-auto">
                  <h3 className="text-sm font-semibold leading-6 text-gray-900">
                    <a href={item.href}>
                      <span className="absolute inset-0" aria-hidden="true" />
                      {item.name}
                    </a>
                  </h3>
                  <p className="mt-2 text-sm leading-6 text-gray-600">{item.description}</p>
                </div>
                <div className="flex-none self-center">
                  <ChevronRightIcon className="h-5 w-5 text-gray-400" aria-hidden="true" />
                </div>
              </li>
            ))}
        </ul>
      </div>
    </div>
  );
};
```

The issue is the list elements are being split.

![image](https://github.com/tailwindlabs/tailwindcss/assets/3826416/d09eeedc-b0f4-4cec-9828-2043a0663693)

I want the "Machine Learning Forecasting. Apply machine learning techniques to the specific task of creating future predictions" to be in the same column.

Please advise.

**Comment by alex-krasikau** [comment, marked as answer]
Hey @greenpau!

`columns` doesn't work well for this case as you have mixed content. You can achieve much more stable results using `grid`. All you need to do is replace `lg:columns-2` with `lg:grid lg:grid-cols-2`.

**Comment by greenpau** [comment]
@alex-krasikau , thank you! 

When I change it like you said, I get a single column.

So I change to have even/odd elements:

```
export const ServiceSelector: React.FC<TrainingContentProps> = (props) => {
  return (
    <div id="getting-started" className="relative isolate">
      <h2 className="sr-only">Getting Started</h2>
      <div className="lg:grid lg:grid-cols-2 lg:gap-16 mx-auto pt-12 mt-16 pb-20 sm:mt-20 lg:px-32">
        <ul role="list" className="-mt-6 divide-y divide-gray-900/5 border-b border-gray-900/5">
          {props.sections &&
            props.sections.map((item, itemIdx) => (
              (itemIdx % 2 === 0) && <li key={`section-even-${itemIdx}`} className="relative flex gap-x-6 py-6 px-12 hover:bg-gray-50">
                ...
              </li>
            ))}
        </ul>
        <ul role="list" className="-mt-6 divide-y divide-gray-900/5 border-b border-gray-900/5">
          {props.sections &&
            props.sections.map((item, itemIdx) => (
              (itemIdx % 2 !== 0) && <li key={`section-even-${itemIdx}`} className="relative flex gap-x-6 py-6 px-12 hover:bg-gray-50">
                ...
              </li>
            ))}
        </ul>
      </div>
    </div>
  );
};
```

However, on a small screen everything gets collapsed and the order of the courses changes.

Any ideas how to best tackle it?

**Reply by alex-krasikau** [reply]
Can you go back to the original code and try adding `[break-inside:avoid]` to the element with `lg:column-2` and let me know about results?

As for grid, could you double-check if you set up the `grid` solution right? It's supposed to create a 2-column layout on big screens - https://play.tailwindcss.com/mdqBJ8k1ov.

**Comment by greenpau** [comment]
@alex-krasikau , I did what you suggested.

![image](https://github.com/tailwindlabs/tailwindcss/assets/3826416/c67d4d24-9cda-4a1f-81e9-53dcf9dca5c8)

Now the issue is that the separator shows up above second column

![image](https://github.com/tailwindlabs/tailwindcss/assets/3826416/b57d9e8a-37c6-471c-a27a-0d319db32059)

**Reply by alex-krasikau** [reply]
Replace `divide` with `border-b` on the `li` element to display a border below each item. If you wish to exclude borders below the 4th and 8th items, using `columns-2` won't suffice; consider switching to a grid solution.
